packages/hardhat/test/integration-tests/crabv2/crab-erc20.ts

```diff
@@ -0,0 +1,215 @@
+import { ethers } from "hardhat"
+import { expect } from "chai";
+import { Contract, BigNumber, providers } from "ethers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/dist/src/signer-with-address";
+import BigNumberJs from 'bignumber.js'
+import { WETH9, MockErc20, Controller, Oracle, WPowerPerp, CrabStrategyV2, ISwapRouter, MockTimelock } from "../../../typechain";
+import { deployUniswapV3, deploySqueethCoreContracts, deployWETHAndDai, addWethDaiLiquidity, addSqueethLiquidity } from '../../setup'
+import { isSimilar, wmul, wdiv, one, oracleScaleFactor } from "../../utils"
+
+BigNumberJs.set({ EXPONENTIAL_AT: 30 })
+
+describe("Crab V2 integration test: ERC20 deposit and withdrawals", function () {
+  const startingEthPrice = 3000
+  const startingEthPrice1e18 = BigNumber.from(startingEthPrice).mul(one) // 3000 * 1e18
+  const scaledStartingSqueethPrice1e18 = startingEthPrice1e18.mul(11).div(10).div(oracleScaleFactor) // 0.3 * 1e18
+  const scaledStartingSqueethPrice = startingEthPrice * 1.1 / oracleScaleFactor.toNumber() // 0.3
+
+
+  const hedgeTimeThreshold = 86400  // 24h
+  const hedgePriceThreshold = ethers.utils.parseUnits('0.01')
+  const auctionTime = 3600
+  const minPriceMultiplier = ethers.utils.parseUnits('0.95')
+  const maxPriceMultiplier = ethers.utils.parseUnits('1.05')
+
+  let provider: providers.JsonRpcProvider;
+  let owner: SignerWithAddress;
+  let depositor: SignerWithAddress;
+  let depositor2: SignerWithAddress;
+  let liquidator: SignerWithAddress;
+  let feeRecipient: SignerWithAddress;
+  let dai: MockErc20
+  let weth: WETH9
+  let positionManager: Contract
+  let uniswapFactory: Contract
+  let swapRouter: Contract
+  let oracle: Oracle
+  let controller: Controller
+  let wSqueethPool: Contract
+  let wSqueeth: WPowerPerp
+  let crabStrategy: CrabStrategyV2
+  let ethDaiPool: Contract
+  let timelock: MockTimelock;
+
+  this.beforeAll("Deploy uniswap protocol & setup uniswap pool", async () => {
+    const accounts = await ethers.getSigners();
+    const [_owner, _depositor, _depositor2, _liquidator, _feeRecipient] = accounts;
+    owner = _owner;
+    depositor = _depositor;
+    depositor2 = _depositor2;
+    liquidator = _liquidator;
+    feeRecipient = _feeRecipient;
+    provider = ethers.provider
+
+    const { dai: daiToken, weth: wethToken } = await deployWETHAndDai()
+
+    dai = daiToken
```

Comment:
can you test with usdc because of difference in decimals?

Reply:
Will try, all other tests use ETH-DAI pool instead of USDC. Should not create any issues with decimals but will change to usdc for sure